### ISSUE TITLE
7(+) seconds of unresponsiveness warning after update VS to 15.6

### ISSUE THREAD

**Issue by nphmuller** [Microsoft/VS-PPT#173]
In Visual Studio 15.6 the unresponsiveness check has been updated to show to part of VS (or extension) that caused it.

Now I keep getting the message `Extension 'Productivity Power Tools 2017 15.0.4' likely caused 7 seconds of unresponsiveness. Disabling it may improve your experience.`. Sometimes with 7 seconds, sometimes a bit more.

Is there something that can be changed to keep this unresponsiveness from happening?

**Comment by justcla** [comment]
Yes, you can uninstall it. That will stop the warnings coming through. :-)

Now I know that might sounds silly. But the PPT extension itself is nothing more than an extension installer. Its sole job is to run at startup and check which extensions from the PPT list are missing, and go download and install them. Once all PPT extensions are installed, the PPT installer has no useful purpose and really should just be uninstalled (or disabled).

That should keep the unresponsiveness should happening.

**Comment by nphmuller** [comment]
@justcla That's fine for me, but isn't there a way to stop the unresponsiveness from happening at all? It probably sounds easier then it is, but isn't it feasible to do the checks in the background, without blocking anything?

**Comment by justcla** [comment]
Good call, Nick. I've started an internal thread to see if we can push the loading to the background.

It currently loads on startup and checks which extensions are installed (on the main thread), then awaits a background task to download and install missing extensions.
We could potentially:
- load the package on an AsyncPackage that runs on the background (to delay package loading), and/or
- check existing extensions on a background thread.

Either of those solutions should reduce UI delays on startup.

@prnadago, what are your thoughts?

**Comment by giggio** [comment]
I also got the warning, and it was on the options page:

![image](https://user-images.githubusercontent.com/334958/37495192-ef6b58f0-288a-11e8-94b3-a9cdae5a155c.png)
